File: temps/mp/src/vdata.js
const Intify = n => Number(n) || 0

export class VData {
  constructor() {
    this.f = 0
    this.delay = 0
    this.te = 0
    this.min = -Infinity
    this.max = Infinity
  }
  Inertia(f) {
    this.f = Math.min(80, Math.abs(Intify(f)))

    return this
  }
  Delay(t) {
    this.delay += Intify(t)

    return this
  }
  Throttle(t) {
    this.te = Intify(t)

    return this
  }
  // 限制最终计算的最大最小值，结合原值，在 width 属性下，类似于限制 min-width max-width
  Limit(min, max) {
    this.min = min
    this.max = max

    return this
  }
}

File: temps/flutter/lib/common/vdata.dart

import 'dart:math' as math;
import './mixin.dart';

class VData {
  double f = 0.0;
  double delay = 0.0;
  double te = 0.0;
  double min = -double.infinity;
  double max = double.infinity;
  
  Inertia(n) {
    f = math.min(80.0, doubleIt(n).abs());

    return this;
  }
  Delay(t) {
    delay += doubleIt(t);

    return this;
  }
  Throttle(t) {
    te = doubleIt(t);

    return this;
  }
  Limit(m, x) {
    min = doubleIt(m);
    max = doubleIt(x);

    return this;
  }
}